src/solver/optimisation/opt_rename_problem.cpp

```diff
@@ -0,0 +1,112 @@
+#include "opt_rename_problem.h"
+#include <sstream>
+void RenameLinkVariable(PROBLEME_ANTARES_A_RESOUDRE* problem,
+                        int var,
+                        Antares::Data::Enum::ExportStructDict structDict,
+                        int ts,
+                        Antares::Data::Enum::ExportStructTimeStepDict timeStepType,
+                        const std::string& origin,
+                        const std::string& destination)
+{
+    auto nvars = problem->NombreDeVariables;
+    if (nvars > var && problem->NomDesVariables[var][0] == '\0')
+    {
+        const auto location = origin + ZONE_SEPARATOR + destination;
+        auto fullName = BuildName(
+          Antares::Data::Enum::toString(structDict),
+          LocationIdentifier(location, Antares::Data::Enum::ExportStructLocationDict::link),
+          TimeIdentifier(ts, timeStepType));
+        problem->NomDesVariables[var] = fullName;
+    }
+}
+
+void RenameZoneVariable(PROBLEME_ANTARES_A_RESOUDRE* problem,
+                        int var,
+                        Antares::Data::Enum::ExportStructDict structDict,
+                        int ts,
+                        Antares::Data::Enum::ExportStructTimeStepDict timeStepType,
+                        const std::string& zone)
+{
+    auto nvars = problem->NombreDeVariables;
+    if (nvars > var)
+    {
+        auto fullName
+          = BuildName(Antares::Data::Enum::toString(structDict),
+                      LocationIdentifier(zone, Antares::Data::Enum::ExportStructLocationDict::area),
+                      TimeIdentifier(ts, timeStepType));
+        problem->NomDesVariables[var] = fullName;
+    }
+}
+void RenameZoneVariable(PROBLEME_ANTARES_A_RESOUDRE* problem,
+                        int var,
+                        Antares::Data::Enum::ExportStructDict structDict,
+                        int ts,
+                        Antares::Data::Enum::ExportStructTimeStepDict timeStepType,
+                        const std::string& zone,
+                        int layerIndex)
+{
+    auto nvars = problem->NombreDeVariables;
+    if (nvars > var)
+    {
+        auto fullName
+          = BuildName(Antares::Data::Enum::toString(structDict),
+                      LocationIdentifier(zone, Antares::Data::Enum::ExportStructLocationDict::area)
+                        + SEPARATOR + "Layer<" + std::to_string(layerIndex) + ">",
+                      TimeIdentifier(ts, timeStepType));
+        problem->NomDesVariables[var] = fullName;
+    }
+}
+void RenameThermalClusterVariable(PROBLEME_ANTARES_A_RESOUDRE* problem,
```

Comment:
**Remark more general than the current file scope**.

RenameThermalClusterVariable(...) and **RenameShortTermStorageVariable(...)** are just about the same.
So only one function is enough.
Moreover, to me, some arguments are useless  for these 2 functions: 
- For these 2 functions, **timeStepType** is always hourly.
- **structDict** can probably avoided (see a bit further in this comment)
- Not sure **problem** is useful, as this function could return the **fullname**, removing the need for this argument. And the function would be more clear. 
  
Instead of a bunch of free functions, we could create a class, this would surely simplify file **opt_construction_matrice_des_contraintes_cas_lineaire.cpp**
- The starting idea is to reduce the number of arguments each function take.
- Another idea is that we have loops over time steps, areas and links, so we could update the time step, the area, ... of an object of this class at the beginning of the loops, so that we would not have to pass them as argument.

In **opt_construction_matrice_des_contraintes_cas_lineaire.cpp**, it could be : 
```C++
void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(...)
{
    // The ProblemeAResoudre now does not need to be passed as argument in any 
    // function or method dealing with variable names    
    VariablesNamer varNamer(problemeHebdo->ProblemeAResoudre); 
   
    for (int pdt = 0; ....)
    {
          varNamer.updateTimeStep(pdt);
          for (int pays = 0; ....)
          {
              varNamer.updateAreaName(problemeHebdo->NomsDesPays[pays]);
              ...
              // Number of arguments reduced because addHydroGeneration already knows things
              // and the complexity is move inside the object / class
              varNamer.addHydroGeneration(varNumber);
              ...
              varNamer.addUnsuppliedEnergy(varNumber);
              ...
          }
    }    
    ....
}

```

Reply:
Moreover, as renaming variables has nothing to do with building constraints, we should try to move this renaming to another place, that is where the variables are built : 
- **opt_construction_variables_optimisees_lineaire.cpp**
- **opt_construction_variables_couts_demarrages.cpp**